holoviews/util/__init__.py

```diff
@@ -502,15 +502,22 @@ def __call__(self, *args, **options):
                 if k not in Store.output_settings.allowed:
                     raise KeyError('Invalid keyword: %s' % k)
             if 'filename' in options:
-                def save_fn(obj, renderer): renderer.save(obj, options['filename'])
-                Store.output_settings.output(line=line, cell=obj, cell_runner=save_fn,
-                                             help_prompt=help_prompt, **options)
-            elif warn:
-                self.warning("hv.output not supplied a filename to export the "
-                             "given object. This call will have no effect." )
-            return obj
-        elif obj is not None:
-            return obj
+                if util.config.future_deprecations:
+                    self.warning('The filename argument of output is deprecated. '
+                                 'Use hv.save instead.')
+
+            def display_fn(obj, renderer):
+                try:
+                    from IPython.display import display
+                except:
+                    return
+
+                # Small hack to avoid printing in IPython terminal
+                if get_ipython().__class__.__name__ != 'TerminalInteractiveShell':
+                    display(obj)
```

Comment:
I don't think we need to go out of our way to make things cleaner in an IPython terminal.  As long as it doesn't do something *too* crazy, I'd skip this check, because users wouldn't be using any of this in such a context normally.